File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/core/handler/array_get_size_handler.rb
require_relative 'abstract_command_handler'

class ArrayGetSizeHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end

  def process(command)
    begin
      if command.payload.length < @required_parameters_count
        raise ArgumentError.new "Array get size parameters mismatch"
      end

      if command.payload[0].is_a? Array
        return get_size_array(command)
      else
        return get_size(command)
      end
    rescue Exception => e
      return e
    end
  end

  def get_size_array(command)
    array = command.payload[0]
    size = 1
    while array.is_a? Array
      size = size * array.length
      array = array[0]
    end
    size
  end

  def get_size(command)
    command.payload[0].length
  end
end

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/ArrayGetSizeHandler.py
from javonet.core.handler.AbstractCommandHandler import AbstractCommandHandler

class ArrayGetSizeHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        try:
            if len(command.payload) != self._required_parameters_count:
                raise Exception("ArrayGetSizeHandler parameters mismatch!")

            array = command.payload[0]
            size = 1
            while(isinstance(array, list)):
                size = size * len(array)
                array = array[0]

            return size


        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None
